### PR TITLE
Feature/close bill and add historic

### PR DESCRIPTION
**Pull request by eccard**

### CODE REVIEW

File: app/src/main/java/com/r5k/contacerveja/data/database/repository/bill/BillsRepo.kt [hunk 1]
```diff
@@ -5,10 +5,14 @@ interface BillsRepo{
...
 
-    fun insertBiil(bill: Bill) : Long
+    fun insertBill(bill: Bill) : Long
 
-    fun loadBiils() : List<Bill>
+    fun loadBills() : List<Bill>
 
     fun loadOpenedBills() : List<Bill>
 
+    fun closeBil(billId: Long) : Int
```
Comment (m4theuslim4) [inline app/src/main/java/com/r5k/contacerveja/data/database/repository/bill/BillsRepo.kt]: closeBill

File: app/src/main/java/com/r5k/contacerveja/data/database/repository/bill/BillsRepository.kt [hunk 2]
```diff
@@ -6,12 +6,19 @@ class BillsRepository @Inject constructor(private val billsDao: BillsDao) : Bill
...
         return this.billsDao.insert(bill)
     }
 
-    override fun loadBiils(): List<Bill> =  billsDao.loadAllBills()
+    override fun loadBills(): List<Bill> =  billsDao.loadAllBills()
 
-    override fun loadOpenedBills(): List<Bill>  = billsDao.loadOpenedBill()
+    override fun loadOpenedBills(): List<Bill>  = billsDao.loadBillWithState(BillState.OPEN.ordinal)
 
+    override fun closeBil(billId: Long): Int {
```
Comment (m4theuslim4) [inline app/src/main/java/com/r5k/contacerveja/data/database/repository/bill/BillsRepository.kt]: closeBill

File: app/src/main/java/com/r5k/contacerveja/ui/historic/view/SimpleDividerItemDecoration.kt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        val right = parent.width - parent.paddingRight -paddinInPx
+
+        val childCount = parent.childCount
+        for (i in 0 until childCount) {
+            val child = parent.getChildAt(i)
+
+            val params = child.layoutParams as RecyclerView.LayoutParams
+
+            val top = child.bottom + params.bottomMargin
+            val bottom = top + mDivider!!.intrinsicHeight
```
Comment (m4theuslim4) [inline app/src/main/java/com/r5k/contacerveja/ui/historic/view/SimpleDividerItemDecoration.kt]: You could declare mDivider with lateinit or lazy and remove "!!" if you sure that this var will never be null. If must be null, you could treat this using ?.let { }

File: app/src/main/java/com/r5k/contacerveja/data/database/repository/bill/BillsRepo.kt (hunk 1 above)
Comment (eccard) [reply app/src/main/java/com/r5k/contacerveja/data/database/repository/bill/BillsRepo.kt]: ok

File: app/src/main/java/com/r5k/contacerveja/ui/historic/view/SimpleDividerItemDecoration.kt (hunk 3 above)
Comment (eccard) [reply app/src/main/java/com/r5k/contacerveja/ui/historic/view/SimpleDividerItemDecoration.kt]: ok

File: app/src/main/java/com/r5k/contacerveja/data/database/repository/bill/BillsRepository.kt (hunk 2 above)
Comment (eccard) [reply app/src/main/java/com/r5k/contacerveja/data/database/repository/bill/BillsRepository.kt]: ok

File: app/src/main/java/com/r5k/contacerveja/util/AppConstants.kt [hunk 4]
```diff
@@ -2,4 +2,7 @@ package com.r5k.contacerveja.util
 
 object AppConstants{
     internal val APP_DB_NAME = "beer_count.db"
+    internal val ACTION_NEW_BILL = "ACTION_ADD_NEW_BILL"
+    internal val ACTION_LOAD_BILL = "ACTION_ADD_LOAD_BILL"
```
Comment (lssalgado) [inline app/src/main/java/com/r5k/contacerveja/util/AppConstants.kt]:
```suggestion
    internal val ACTION_LOAD_BILL = "ACTION_LOAD_BILL"
```

**Pull request merged**